storage/database/db_migration.go

```diff
@@ -0,0 +1,120 @@
+// Modifications Copyright 2020 The klaytn Authors
+// Copyright 2015 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package database
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	dbMigrationFetchNum = 500
+)
+
+// StartDBMigration migrates a DB to a different kind of DB.
+// (e.g. LevelDB -> BadgerDB, LevelDB -> DynamoDB)
+//
+// This feature uses Iterator. A src DB should have implementation of Iteratee to use this function.
+// Do not use db migration while a node is executing.
+func (dbm *databaseManager) StartDBMigration(dstdbm DBManager) error {
+	// settings for quit signal from os
+	sigQuit := make(chan os.Signal, 1)
+	signal.Notify(sigQuit,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+
+	// TODO enable for all dbs
+	srcDB := dbm.getDatabase(MiscDB) // first DB
+	dstDB := dstdbm.getDatabase(MiscDB)
+
+	// create src iterator and dst batch
+	srcIter := srcDB.NewIterator()
+	dstBatch := dstDB.NewBatch()
+
+	// vars for log
+	start := time.Now()
+	iterateNum, fetchedTotal, fetched := 0, 0, 1
+
+loop:
+	for fetched > 0 {
+		// fetch keys and values
+		var keys, vals [][]byte
+		iterStart := time.Now()
+		keys, vals, fetched = iterateDB(srcIter, dbMigrationFetchNum)
+		elapsedFetchTime := time.Since(iterStart)
+		fetchedTotal += fetched
+
+		// write fetched keys and values to DB
+		putStart := time.Now()
+		for i := 0; i < fetched; i++ {
+			// If dstDB is dynamoDB, Put will Write when the number items reach dynamoBatchSize.
+			if err := dstBatch.Put(keys[i], vals[i]); err != nil {
+				return errors.Wrap(err, "failed to put batch")
+			}
+		}
+		logger.Info("DB migrated", "iterNum", iterateNum, "fetched", fetched,
+			"elapsedFetch", elapsedFetchTime, "elapsedPut", time.Since(putStart), "elapsedIter", time.Since(iterStart),
+			"fetchedTotal", fetchedTotal, "elapsedTotal", time.Since(start))
```

Comment:
Isn't this line printed too much? I think `dbMigrationFetchNum` items can be processed less than 1 second on average.

Reply:
Yes, one iteration (fetching 500 entries) takes about 3ms seconds. I think it is okay to print a long log. It is the only thing this process is doing. Also, it is useful for monitoring.

Reply:
I reflected your comments. PTAL